### DISCUSSION TITLE
Iterate through the JSON numeric index

### DISCUSSION THREAD

**Discussion by Kolhun** [nlohmann/json#4148] [Q&A]
Perhaps this question is not directly related to the library, but I have a question, how to sort the contents of an array using a numeric index (int)? As in the example below. The code below is supposed to work but data[row].value() is clearly not the correct construction

```c
 json data = dataResult["data"];

ImGuiListClipper clipper;
clipper.Begin(data.size());
while (clipper.Step())
{
    for (int row = clipper.DisplayStart; row < clipper.DisplayEnd; row++)//for (auto row = data.begin(); row != data.end(); ++row)
    {
        ImGui::TableNextRow();
        for (int column = 0; column < 4; column++)
        {
            ImGui::TableSetColumnIndex(column);
            json dataResultList = data[row].value();
```
![image](https://github.com/nlohmann/json/assets/36815965/b072d7b4-e66b-43db-a6a1-fed4ee2104e8)

**Comment by nlohmann** [comment]
Same fix as for the sibling discussion #4156: use data[row] directly to index into a json array by integer position — data[row].value() is not valid because plain indexing already returns the element, not an iterator you need to call .value() on. If you need it sorted by some numeric field rather than array position, use std::sort/std::stable_sort on data (a basic_json array behaves like a standard container of json elements) with a custom comparator that reads the field you want to sort by.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
